phpBB/phpbb/ban/manager.php

```diff
@@ -0,0 +1,270 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\ban;
+
+/**
+ * Class for managing bans
+ */
+class manager
+{
+	/**
+	 * Database object
+	 * @var \phpbb\db\driver\driver_interface
+	 */
+	protected $db;
+
+	/**
+	 * Event dispatcher object
+	 * @var \phpbb\event\dispatcher_interface
+	 */
+	protected $dispatcher;
+
+	/**
+	 * Log object
+	 * @var \phpbb\log\log_interface
+	 */
+	protected $log;
+
+	/**
+	 * Ban type map
+	 * @var array
+	 */
+	protected $type_map;
+
+	/**
+	 * User object
+	 * @var \phpbb\user
+	 */
+	protected $user;
+
+	public function __construct($ban_types, \phpbb\db\driver\driver_interface $db, \phpbb\event\dispatcher_interface $dispatcher, \phpbb\log\log_interface $log, \phpbb\user $user)
+	{
+		$this->db = $db;
+		$this->dispatcher = $dispatcher;
+		$this->log = $log;
+		$this->user = $user;
+
+		$this->fill_type_map($ban_types);
+	}
+
+	/**
+	 * Fill ban type map
+	 *
+	 * @param \phpbb\di\service_collection
+	 */
+	protected function fill_type_map($types)
+	{
+		$this->type_map = array();
+
+		foreach ($types as $type)
+		{
+			$this->type_map[$type->get_type()] = $type;
```

Comment:
avoid loading the services it's better to keep the collection and use the service name for $mode